### ISSUE TITLE
False positive UnnecessaryParentheses on constructors with lambdas

### ISSUE THREAD

**Issue by neonailol** [arturbosch/detekt#990]
Detekt reports UnnecessaryParentheses  for the secondary constructor in the following code for class `Clazz`:

```kotlin

typealias KBiFunc<X, Y, Z> = (X, Y) -> Z

interface Func<in X : Any, out Y : Any> {
    fun apply(input: X): Y
}

interface BiFunc<in X : Any, in Y : Any, out Z : Any> {
    fun apply(
        first: X,
        second: Y
    ): Z
}

class Clazz<in X : Any, out Y : Any>(
    private val func: KBiFunc<X, Boolean, Y>
) : Func<X, Y> {

    constructor(
        func: BiFunc<X, Boolean, Y>
    ) : this(
        { first, second -> func.apply(first, second) }
    )

    override fun apply(input: X): Y = func(input, true)
}

```

I am using version `1.0.0.RC7-3`

**Comment by arturbosch** [comment]
Thanks for reporting. I could reproduce the bug with your code.
Is it okay if I use your code for a test case?

**Comment by schalkms** [comment]
This might be related to #927

**Comment by neonailol** [comment]
@arturbosch yes, of course
